Alpaca.Markets/SockClient.cs

```diff
@@ -82,54 +81,19 @@ public sealed partial class SockClient : IDisposable
         /// </summary>
         public event Action<ITradeUpdate> OnTradeUpdate;
 
-        /// <summary>
-        /// Occured when stream successfully connected.
-        /// </summary>
-        public event Action<AuthStatus> Connected;
-
         /// <summary>
         /// Occured when any error happened in stream.
         /// </summary>
         public event Action<Exception> OnError;
 
         /// <summary>
-        /// Opens connection to Alpaca streaming API.
-        /// </summary>
-        /// <returns>Waitable task object for handling action completion in asynchronous mode.</returns>
-        public Task ConnectAsync()
-        {
-            return _webSocket.OpenAsync();
-        }
-
-        /// <summary>
-        /// Closes connection to Alpaca streaming API.
+        /// Occured when stream successfully connected.
         /// </summary>
-        /// <returns>Waitable task object for handling action completion in asynchronous mode.</returns>
-        public Task DisconnectAsync()
-        {
-            return _webSocket.CloseAsync();
-        }
-
-        /// <inheritdoc />
-        public void Dispose()
-        {
-            if (_webSocket == null)
-            {
-                return;
-            }
-
-            _webSocket.Opened -= handleOpened;
-            _webSocket.Closed -= handleClosed;
-
-            _webSocket.DataReceived -= handleDataReceived;
-            _webSocket.Error -= handleError;
-
-            _webSocket.Dispose();
-        }
+        public event Action<AuthStatus> Connected;
 
-        private void handleOpened()
+        override private protected JsonAuthRequest getAuthRequest()
```

Comment:
This access modifier (`private protected`) is new for C# 7.2 and can cause build problems for some clients who has older compilers. I recommend to use only C#7.0 features for now (and replace all `private protected` with `internal`) or change .csproj file and enable `latest` compiler option.